backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/course/domain/CourseExecution.java

```diff
@@ -222,4 +230,21 @@ public int getNumberOfQuestions() {
                 .filter(user -> user.getRole().equals(User.Role.STUDENT))
                 .collect(Collectors.toSet());
     }
+
+    public Set<Topic> findAvailableTopics() {
+        Set<Topic> availableTopics = new HashSet<>();
+
+        getAvailableAssessments().forEach(assessment -> availableTopics.addAll(assessment.getTopics()));
+
+        return availableTopics;
+    }
+
+    public List<Question> filterByTopicsTournament(List<Question> availableQuestions, StatementTournamentCreationDto quizDetails, List<Integer> availableTopicsIds) {
```

Comment:
This method does not use anything from this class...
Need to read where it is invoked to decide where it should be

Reply:
Done